lms/djangoapps/course_api/blocks/transformers/milestones.py

```diff
@@ -113,17 +115,11 @@ def add_special_exam_info(self, block_key, block_structure, usage_info):
         """
         For special exams, add the special exam information to the course blocks.
         """
-        special_exam_attempt_context = None
-        try:
-            # Calls into edx_proctoring subsystem to get relevant special exam information.
-            # This will return None, if (user, course_id, content_id) is not applicable.
-            special_exam_attempt_context = get_attempt_status_summary(
-                usage_info.user.id,
-                str(block_key.course_key),
-                str(block_key)
-            )
-        except ProctoredExamNotFoundException as ex:
-            log.exception(ex)
+        special_exam_attempt_context = self._generate_special_exam_attempt_context(
```

Comment:
For now I've created a helper method, but a larger refactoring question I have is:
Should I create a util function that both milestones and special exams can use? If so where could this go?
And an even bigger question: should I even be making the change in special exams, or should this be a different ticket/separate work? It looks like the milestones.py is what is actually changing the frontend, so I'm not sure how special exams is currently being used.